src/nfa/nfa.cc

```diff
@@ -1198,7 +1184,7 @@ Nfa Mata::Nfa::determinize(
                 }
                 worklist.emplace_back(std::make_pair(Tid, T));
             }
-            result.transition_relation[Sid].push_back(Move(currentSymbol, Tid));
+            result.delta[Sid].push_back(Move(currentSymbol, Tid));
```

Comment:
push_back is dangerous, can break sortedness. 
Perhaps use insert instead, which tests whether it can push_back, and if not, it finds the place to insert and insrts. Also, it should add to a move with the same symbol which was already there, if it was already there. It should not addd the second move with the same symbol.